server/src/main/java/org/apache/iotdb/db/query/control/QueryFileManager.java

```diff
@@ -77,8 +74,9 @@ private void addUsedFilesForQuery(long queryId, List<TsFileResource> resources)
       // this file may be deleted just before we lock it
       if (tsFileResource.isDeleted()) {
         Map<Long, Set<TsFileResource>> pathMap = !isClosed ? unsealedFilePathsMap : sealedFilePathsMap;
-        pathMap.get(queryId).remove(tsFileResource);
-        FileReaderManager.getInstance().decreaseFileReaderReference(tsFileResource, isClosed);
+        if (pathMap.get(queryId).remove(tsFileResource)) {

```

Comment:
what If does not contain the tsfileResource, is it a bug?

Reply:
Nope, It may have been decreased because of it's deleted, but another thread may also enter into the same block, it also remove it(although it's an empty operation, the resource has been removed by another thread, so the remove operation return false), in this case, the second thread should not decrease it again.